docker_deploy/config.go

```diff
@@ -114,3 +117,45 @@ func StringSliceEqual(a, b []string) bool {
 	}
 	return true
 }
+
+// mapsEqual checks if two maps are equal
+func mapsEqual(a, b map[string]interface{}) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	for key, valueA := range a {
+		valueB, ok := b[key]
+		if !ok {
+			return false
+		}
+
+		// Different Types
+		if reflect.TypeOf(valueA) != reflect.TypeOf(valueB) {
+			return false
+		}
+
+		// Different Values
+		switch valueA := valueA.(type) {
```

Comment:
you shouldn't need this switch at all, [reflect.deepEqual](https://pkg.go.dev/reflect#DeepEqual) should handle all these cases for you.